x/pocketcore/keeper/keeper.go

```diff
@@ -61,27 +61,43 @@ func (k Keeper) ConvertState(ctx sdk.Ctx) {
 }
 
 func (k Keeper) ConsensusParamUpdate(ctx sdk.Ctx) *abci.ConsensusParams {
+	return k.consensusBlockSizeParamUpdate(ctx)
+}
+
+func (k Keeper) consensusBlockSizeParamUpdate(ctx sdk.Ctx) *abci.ConsensusParams {
+	previousBlockCtx, err := ctx.PrevCtx(ctx.BlockHeight() - 1)
+	if err != nil {
+		ctx.Logger().Error("failed to get previous block context")
+		return &abci.ConsensusParams{}
+	}
+
 	currentHeightBlockSize := k.BlockByteSize(ctx)
-	//If not 0 and different update
-	if currentHeightBlockSize > 0 {
-		previousBlockCtx, _ := ctx.PrevCtx(ctx.BlockHeight() - 1)
-		lastBlockSize := k.BlockByteSize(previousBlockCtx)
-		if lastBlockSize != currentHeightBlockSize {
-			//not go under default value
-			if currentHeightBlockSize < types.DefaultBlockByteSize {
-				return &abci.ConsensusParams{}
-			}
-			return &abci.ConsensusParams{
-				Block: &abci.BlockParams{
-					MaxBytes: currentHeightBlockSize,
-					MaxGas:   -1,
-				},
-				Evidence: &abci.EvidenceParams{
-					MaxAge: 50,
-				},
-			}
-		}
+	// If it's 0, we're using the default value from genesis (i.e. unset)
+	if currentHeightBlockSize == 0 {
+		return &abci.ConsensusParams{}
 	}
 
-	return &abci.ConsensusParams{}
+	lastBlockSize := k.BlockByteSize(previousBlockCtx)
+	// If the block size is unchanged, return empty params
+	if lastBlockSize == currentHeightBlockSize {
+		return &abci.ConsensusParams{}
+	}
+
+	// DISCUSS: Should we floor the block size to the genesis default value?
+	// if currentHeightBlockSize < types.DefaultBlockByteSize {
```

Comment:
Any reason why we removed this? Looks like this will allow us to decrease block size

Reply:
Primarily to have a small number in the test (and localnet in the next PR with instructions I'll be pushing), so it's easy to fill up a block in simulation.

Left a `DISCUSS` comment to discuss in the PR. Given that it's ACLed, do you see any issue with being able to lower it?

Reply:
Added a hack for this using `TestMode`